File: output/csharp/XPack/MachineLearning/Job/Config/TimingStats.cs
using Nest.Internal;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.XPack {

	public class TimingStats  {
		
		[DataMember(Name="average_bucket_processing_time_ms")]
		public double AverageBucketProcessingTimeMs { get; set; }


		[DataMember(Name="bucket_count")]
		public long BucketCount { get; set; }


		[DataMember(Name="exponential_average_bucket_processing_time_ms")]
		public double ExponentialAverageBucketProcessingTimeMs { get; set; }


		[DataMember(Name="exponential_average_bucket_processing_time_per_hour_ms")]
		public double ExponentialAverageBucketProcessingTimePerHourMs { get; set; }


		[DataMember(Name="job_id")]
		public string JobId { get; set; }


		[DataMember(Name="maximum_bucket_processing_time_ms")]
		public double MaximumBucketProcessingTimeMs { get; set; }


		[DataMember(Name="minimum_bucket_processing_time_ms")]
		public double MinimumBucketProcessingTimeMs { get; set; }

	}
}

File: specification/specs/x_pack/machine_learning/job/config/TimingStats.ts
/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

class TimingStats {
  average_bucket_processing_time_ms: double
  bucket_count: long
  exponential_average_bucket_processing_time_ms: double
  exponential_average_bucket_processing_time_per_hour_ms: double
  job_id: string
  maximum_bucket_processing_time_ms: double
  minimum_bucket_processing_time_ms: double
}
